### PR TITLE
chore(tools): Use PR number in changeset titles when Issue is absent

### PR DESCRIPTION
**Pull request by abernix**
Previously, it was just leaving it off, but it left things a bit
unharmonic/off-balance in terms of the resulting changeset template.  I'd
also just been manually adding those during the release process thus far, so
figured I'd just automate it now.

Also added a lot of tests and fixed a couple bugs with trailing spaces.

Relates to #https://github.com/apollographql/router/issues/2261

### CODE REVIEW

**Review by o0Ignition0o — APPROVED** [review]

File: xtask/Cargo.toml [hunk 1]
```diff
@@ -41,3 +41,6 @@ tokio = "1.25.0"
 which = "4"
 zip = { version = "0.6", default-features = false }
 walkdir = "2.3.2"
+
+[dev-dependencies]
+insta = { version = "1.26.0", features = ["json", "redactions", "yaml"] }
```
[the remark this answers is not in the document]
Comment (Geal) [reply xtask/Cargo.toml:46]: more dependencies, yay

File: xtask/src/commands/changeset/mod.rs [hunk 2]
```diff
@@ -66,15 +66,29 @@ struct TemplateContext {
 
 const REPO_WITH_OWNER: &'static str = "apollographql/router";
 
-const EXAMPLE_TEMPLATE: &'static str = "### {title} {{ for issue in issues -}}
-([Issue #{issue.number}]({issue.url})){{ if not @last }}, {{ endif }}
-{{- endfor }}
+const EXAMPLE_TEMPLATE: &'static str = "### { title }
+{{- if issues -}}
+  {{- if issues }} {{ endif -}}
```
Comment (Geal) [inline xtask/src/commands/changeset/mod.rs:71]:
```suggestion
```
shouldn't this be removed?

**Review by Geal — APPROVED** [review]

File: xtask/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (abernix) [reply xtask/Cargo.toml:46]: At least I didn't add `openssl` back. 😉

File: xtask/src/commands/changeset/mod.rs (hunk 2 above)
Comment (abernix) [reply xtask/src/commands/changeset/mod.rs:71]: Good eye and question, but no and hard to comment why it exists. It actually allows me to use a nice formatted structure for this template, while ensuring that there will be a space **preserved** there at all times.

**Pull request merged**